### PR TITLE
LPS-118317 create unit test for the formatBigDecimal method

### PR DESCRIPTION
**Pull request by carolmariaabb**
Related issue: https://issues.liferay.com/browse/LPS-118317

### CODE REVIEW

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by natocesarrego** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessor.java [hunk 1]
```diff
@@ -179,6 +181,37 @@ else if (ddmFormInstanceReportEvent.equals(
 		return jsonObject;
 	}
 
+	protected String formatBigDecimal(BigDecimal bigDecimal) {
+		String bigDecimalString = bigDecimal.toString();
+
+		String[] bigDecimalStringParts = bigDecimalString.split("\\.");
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessor.java:187]: You learned from Brian, @marcosapmf. 😄 👏

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/utils/numeric.es.js [hunk 2]
```diff
@@ -59,18 +59,22 @@ export function formatNumber(number, delimit) {
 
 	const formattedDecimal = formattedNumberParts[1];
 
-	const formattedInteger = numeral(
-		parseInt(formattedNumberParts[0], 10)
-	).format('0[,]0');
+	const formattedInteger = formattedNumberParts[0].replace(
+		/\B(?=(\d{3})+(?!\d))/g,
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/resources/META-INF/resources/js/utils/numeric.es.js:48]: @marcosapmf, please add this regex to a constant.

cc @carolmariaabb

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessorTest.java [hunk 3]
```diff
@@ -251,6 +251,64 @@ public void testProcessDDMFormInstanceReportWithExistingData()
 		Assert.assertEquals("4", summaryJSONObject.getString("sum"));
 	}
 
+	@Test
+	public void testProcessDDMFormInstanceReportWithLongFieldValue()
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/test/java/com/liferay/dynamic/data/mapping/internal/report/NumericDDMFormFieldTypeReportProcessorTest.java:273]: Great job, @marcosapmf! In a very near future, I'll propose to you guys always add a test for every fixed bug.

**Pull request closed without merging**